jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/VertexScoringAlgorithm.java

```diff
@@ -0,0 +1,67 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* ------------------------------
+* VertexScoringAlgorithm.java
+* ------------------------------
+* (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+*
+* Original Author:  Dimitrios Michail
+* Contributor(s):   -
+*
+* $Id$
+*
+* Changes
+* -------
+* 14-Aug-2016 : Initial revision (DM);
+*/
+package org.jgrapht.alg.interfaces;
+
+import java.util.Map;
+
+/**
+ * An interface for all algorithms which assign scores to vertices of a graph.
+ * 
+ * @param <V> the vertex type
+ * @param <D> the score type
+ * 
+ * @author Dimitrios Michail
+ * @since August 2016
+ */
+public interface VertexScoringAlgorithm<V, D>
+{
```

Comment:
I agree that it is now an overkill. 

However, there are a lot of vertex scoring methods that are used in network analysis (degree centrality, closeness centrality, betweenness centrality, eigenvector centrality, PageRank, etc.). The [wikipedia](https://en.wikipedia.org/wiki/Centrality) article is very informative.

Perhaps if someone in the future implemented another vertex scoring algorithm, the interface would make more sense. JUNG has a few algorithms which compute vertex scores, all implementing a similar interface.